apps/prairielearn/assets/scripts/behaviors/popover.ts

```diff
@@ -1,22 +1,66 @@
+import { on } from 'delegated-events';
 import { observe } from 'selector-observer';
 
 import { onDocumentReady } from '@prairielearn/browser-utils';
 
+const openPopovers = new Set<any>();
+
+function getPopoverContainerForTrigger(trigger: HTMLElement): HTMLElement | null {
+  const popoverId = trigger.getAttribute('aria-describedby');
+  if (!popoverId) return null;
+
+  const popoverContainer = document.querySelector<HTMLElement>(`#${popoverId}`);
+  return popoverContainer;
+}
+
+function getPopoverTriggerForContainer(container: HTMLElement): HTMLElement | null {
+  const popoverId = container.getAttribute('id');
+  if (!popoverId) return null;
+
+  const popoverTrigger = document.querySelector<HTMLElement>(`[aria-describedby="${popoverId}"]`);
+  return popoverTrigger;
+}
+
+observe('[data-toggle="popover"]', {
+  add(el) {
+    $(el).popover({ sanitize: false });
+  },
+  remove(el) {
+    $(el).popover('dispose');
```

Comment:
I don't think this is doing what it's supposed to do. I tried commenting out the call to `popover('hide')` in the assessment instances client asset to check if this function would be enough to dispose of the popover, and this function is called, but the popover is not removed. I believe by the time this is called the link between the element and its popover has been removed already. In any case, I tested calling:

```ts
    const container = getPopoverContainerForTrigger(el as HTMLElement);
    if (container) {
      $(container).popover('dispose');
    }
```
and it more reliably closed the popover.

Reply:
Can you provide step-by-step reproduction instructions for whatever problem you're seeing?

Reply:
Let me know if this was what you were testing, but I was able to reproduce a problem when enabling "Auto Refresh" on the assessment instances page, opening some popovers, and then waiting for a refresh. Because the DOM is recreated from scratch by `bootstrap-table`, the trigger elements are destroyed, but the popovers are not, and the "Cancel" buttons no longer work once the triggers are gone.

Reply:
I pushed what should be a fix to this here: b1ae552. Essentially, we'll forcefully remove the popover container when its trigger goes away, even if Bootstrap "forgot" about the link.

This might get fixed in BS5 once we're no longer relying on jQuery.

Reply:
This did indeed take care of the issue. That said, keeping in mind conflicting changes between this and #10437, you might want to add `[data-bs-toggle-popover]` to the observable selectors for this action once that PR is merged (depending on which one is merged first). If you do that, the code below can be removed:

https://github.com/PrairieLearn/PrairieLearn/blob/b1ae552540ba88a14a79b3594de7150ce326cd7a/apps/prairielearn/assets/scripts/instructorAssessmentInstancesClient.ts#L106-L113

Reply:
Just saw your other comment, please ignore this one.